### DISCUSSION TITLE
a way to append client_storage without using set and does not delete previous data

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#1090] [Q&A]
### Question

Is there a way to add data?  but does not overwrite old data.  let's say I have a case.  with the school key.  and has data ['room','lamp','teacher'] .  and I want to add data into the key scholl.  without having to remove the previous data.  suppose I want to add more data ['milk,'sauce','monday'] .  so for client_storage that has a scholl key.  will have data ['room','lamp','teacher','milk','sauce','monday']

**is there a way like this?**
**page.client_storage.append("mycart",data)**

### Code sample

```python
data = [
			{"name":itemselectname.value},
			{"price":itemselectprice.value},
			{"image":itemselectimage.src},
			{"you_buy_pcs":youselectprice.value},
		]
		page.client_storage.set("scholl",data)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by iron3oxide** [comment]
You could simply get the values for the key first (in your case you would then have a list `['room','lamp','teacher']`) and then append `['milk,'sauce','monday']` to that, resulting in the intended value `['room','lamp','teacher','milk','sauce','monday']`, which you then write to client storage again.

**Reply by iron3oxide** [reply]
"Append" may have been the wrong word, adding two lists together works differently of course, but I hope you get the gist.
